aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalogInteg.java

```diff
@@ -0,0 +1,271 @@
+    glueCatalog.dropNamespace(Namespace.of(namespace));
+    AssertHelpers.assertThrows("namespace should not exist after deletion",
+        EntityNotFoundException.class,
+        () -> GLUE.getDatabase(GetDatabaseRequest.builder().name(namespace).build()));
+  }
+
+  @Test
+  public void testCreateTable() {
+    // create ns
+    String namespace = getRandomName();
+    NAMESPACES.add(namespace);
+    glueCatalog.createNamespace(Namespace.of(namespace));
+    // exists
+    Assert.assertTrue(glueCatalog.namespaceExists(Namespace.of(namespace)));
+    // list table should have nothing
+    Assert.assertTrue(glueCatalog.listTables(Namespace.of(namespace)).isEmpty());
+    // create table
+    Schema schema = new Schema(Types.NestedField.required(1, "c1", Types.StringType.get(), "c1"));
+    PartitionSpec partitionSpec = PartitionSpec.builderFor(schema).build();
+    String tableName = getRandomName();
+    glueCatalog.createTable(TableIdentifier.of(namespace, tableName), schema, partitionSpec);
+    GetTableResponse response = GLUE.getTable(GetTableRequest.builder()
+        .databaseName(namespace).name(tableName).build());
+    Assert.assertEquals(namespace, response.table().databaseName());
+    Assert.assertEquals(tableName, response.table().name());
+    Assert.assertEquals(BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH),
+        response.table().parameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP));
+    Assert.assertTrue(response.table().parameters().containsKey(BaseMetastoreTableOperations.METADATA_LOCATION_PROP));
+    String metaLocation = response.table().parameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+    String key = metaLocation.split(BUCKET)[1].substring(1);
+    S3.headObject(HeadObjectRequest.builder().bucket(BUCKET).key(key).build());
+    // exists
+    Assert.assertTrue(glueCatalog.tableExists(TableIdentifier.of(namespace, tableName)));
+    // load
+    Table table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    Assert.assertEquals(partitionSpec, table.spec());
+    Assert.assertEquals(schema.toString(), table.schema().toString());
+    // list table should show
+    List<TableIdentifier> tables = glueCatalog.listTables(Namespace.of(namespace));
+    Assert.assertEquals(1, tables.size());
+    Assert.assertEquals(TableIdentifier.of(namespace, tableName), tables.get(0));
+    // duplicate
+    AssertHelpers.assertThrows("should not create table with the same name",
+        AlreadyExistsException.class,
+        () -> glueCatalog.createTable(TableIdentifier.of(namespace, tableName), schema, partitionSpec));
+    // bad name
+    AssertHelpers.assertThrows("should not create table with bad name",
+        IllegalArgumentException.class,
+        () -> glueCatalog.createTable(TableIdentifier.of(namespace, "table-1"), schema, partitionSpec));
+    // drop
+    glueCatalog.dropTable(TableIdentifier.of(namespace, tableName));
```

Comment:
do we want to/is it easy to test the `purge` parameter?

Reply:
yes I can add that test